pkg/api/documentsschema/documentsschema.proto

```diff
@@ -0,0 +1,252 @@
+syntax = "proto3";
+
+package immudb.documentsschema;
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+import "schema.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option go_package = "github.com/codenotary/immudb/pkg/api/documentsschema";
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  base_path: "/api/v2",
+  info: {
+    title: "immudb REST API v2";
+    description: "Objects storage API"
+  };
+  security_definitions: {
+    security: {
+      key: "bearer"
+      value: {
+        type: TYPE_API_KEY
+        in: IN_HEADER
+        name: "grpc-metadata-sessionid"
+        description: "Session token: <token>"
+      }
+    }
+  }
+  security: {
+    security_requirement: {
+      key: "bearer"
+    }
+  }
+};
+
+message DocumentInsertRequest {
+  string collection = 1;
+  repeated google.protobuf.Struct document = 2;
+}
+
+message DocumentInsertResponse {
+  schema.VerifiableTx proof = 1;
+}
+
+
+enum QueryOperator {
+  EQ = 0;
+  GT = 1;
+  GTE = 2;
+  LT = 3;
+  LTE = 4;
+  LIKE = 5;
+}
+
+message IndexValue {
+  oneof kind {
+    google.protobuf.NullValue null_value = 1;
+    double number_value = 2;
+    string string_value = 3;
+    bool bool_value = 4;
+  }
+}
+
+message DocumentQuery {
+  string field = 1;
+  QueryOperator operator = 2;
+  google.protobuf.Value value = 3;
+}
+
+
+message DocumentSearchRequest {
+  string collection = 1;
+  repeated DocumentQuery query = 2;
+  uint32 page = 3;
+  uint32 perPage = 4;
+}
+
+message DocumentSearchResponse {
+  repeated google.protobuf.Struct results = 1;
+  uint32 page = 2;
+  uint32 perPage = 3;
+  uint32 entriesLeft = 4;
+}
+
+
+message DocumentProofRequest {
+  string collection = 1;
+  map<string, IndexValue> primaryKeys = 2;
+  int64 atRevision = 3;
+}
+
+
+message DocumentProofResponse {
+  schema.VerifiableTx proof = 1;
+}
+
+
+message DocumentAuditRequest {
+  string collection = 1;
+  map<string, IndexValue> primaryKeys = 2;
+  uint32 page = 3;
+  uint32 perPage = 4;
+}
+
+
+message DocumentAuditResponse {
+  repeated DocumentAudit results = 1;
+  uint32 page = 2;
+  uint32 perPage = 3;
+  uint32 entriesLeft = 4;
+}
+
+message DocumentAudit {
+  google.protobuf.Struct value = 1;
+  uint64 transactionID = 2;
+}
+
+
+message IndexOption {
+  IndexType type = 1;
+}
+
+enum IndexType {
+  DOUBLE = 0;
+  INTEGER = 1;
+  STRING = 2;
+}
+
+message CollectionCreateRequest {
+  string name = 1;
+  map<string, IndexOption> primaryKeys = 2;
+  map<string, IndexOption>  indexKeys = 3;
```

Comment:
to be aware of: it won't be possible to define compound indexes

Reply:
To discuss, currently not important i believe